addons/account/models/res_bank.py

```diff
@@ -47,15 +54,20 @@ def build_qr_code_vals(self, amount, free_communication, structured_communicatio
         return None
 
     def build_qr_code_url(self, amount, free_communication, structured_communication, currency, debtor_partner, qr_method=None, silent_errors=True):
-        candidate_method, amount, currency, debtor_partner, free_communication, structured_communication = \
-            self.build_qr_code_vals(amount, free_communication, structured_communication, currency, debtor_partner, qr_method, silent_errors)
-        return self._get_qr_code_url(candidate_method, amount, currency, debtor_partner, free_communication, structured_communication)
-
+        vals = self._build_qr_code_vals(amount, free_communication, structured_communication, currency, debtor_partner, qr_method, silent_errors)
+        if vals:
+            qr_method, amount, currency, debtor_partner, free_communication, structured_communication = \
+                vals['qr_method'], vals['amount'], vals['currency'], vals['debtor_partner'], vals['free_communication'], vals['structured_communication'],
+            return self._get_qr_code_url(qr_method, amount, currency, debtor_partner, free_communication, structured_communication)
```

Comment:
Isn't this more readable?

```suggestion
            return self._get_qr_code_url(
                vals['qr_method'], 
                vals['amount'], 
                vals['currency'], 
                vals['debtor_partner'], 
                vals['free_communication'], 
                vals['structured_communication'],
            )

Reply:
Big apologies, seems I am not fully awake :sweat_smile: Thanks